### PR TITLE
check.rb - update for newer Rubies / OpenSSL

### PR DESCRIPTION
**Pull request by MSP-Greg**

### CODE REVIEW

**Comment by MSP-Greg** [comment]
@indirect

> my fork is actually the original, now transferred to rubygems.

However it happened, thanks for writing the code.

But, I think the short URI still goes to your fork, at least that's what my browser is showing?  IOW

https://git.io/vQhWq =>
https://raw.githubusercontent.com/indirect/ruby-ssl-check/master/check.rb

**Comment by MSP-Greg** [comment]
@indirect

This seems to be working, I added GitHub Actions and it does pass in my fork:
https://github.com/MSP-Greg/ruby-ssl-check/actions/runs/109649384

Note that one does have to open the 'run check.rb' step to check it.  Actions appears to be disabled here...

File: check.rb [hunk 1]
```diff
@@ -65,14 +66,11 @@ def error_reason(error)
...
-    bundler_uri = Bundler::URI(uri)
-  else
-    bundler_uri = uri
-  end
-  Bundler::Fetcher.new(Bundler::Source::Rubygems::Remote.new(uri)).send(:connection).request(uri)
+  b_uri = defined?(Bundler::URI) ? Bundler::URI(uri.to_s) : uri
+  Bundler::Fetcher.new(Bundler::Source::Rubygems::Remote.new(b_uri)).send(:connection).request(b_uri)
   bundler_status = "success ✅"
 rescue => error
+puts error.class, error.message
```
Comment (indirect) [inline check.rb:73]:
```suggestion
```

File: check.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@
   require 'rubygems'
   # Try for Bundler version
   require 'bundler'
+  require 'bundler/vendor/uri/lib/uri'
```
Comment (indirect) [inline check.rb:29]: This is autoloaded by `require 'bundler'`, isn't it?

Comment (MSP-Greg) [reply check.rb:29]: You're correct.  Sorry, I had too many console windows open...

Sorry about the debug print.  Note to self - debug prints for when it's broken disappear once you've fixed it...

Comment (indirect) [reply check.rb:29]: No worries, I'll just take it out.

```suggestion
```

**Comment by MSP-Greg** [comment]
@indirect

JFYI, rubygems.org doesn't support TLSv1.3 yet.  Thanks, sorry for the dumb ass code I left in..

**Comment by indirect** [comment]
@MSP-Greg Fastly hasn't yet enabled TLS 1.3 globally: https://docs.fastly.com/products/tls-service-options#tls-13-and-0-rtt. We could opt in, but I'm not sure if that's much of a win yet, as clients are still upgrading.

**Comment by MSP-Greg** [comment]
> We could opt in

Maybe try and see if it helps.  Both OpenSSL 1.0.2 & 1.1.0 are EOL, so I would hope that most servers are using 1.1.1 (which is the first version with TLSv1.3).  It should downgrade to TLSv1.2 if the client can't run TLSv1.3...

As the Fastly doc mentions, TLSv1.3 is more efficient than TLSv1.2...

**Pull request merged**